x-pack/plugins/saved_objects_tagging/public/components/assign_flyout/lib/parse_query.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Query } from '@elastic/eui';
+
+export interface ParsedQuery {
+  /**
+   * Combined value of the term clauses
+   */
+  queryText?: string;
+  /**
+   * The values of the `type` field clause (that are populated by the `type` filter)
+   */
+  selectedTypes?: string[];
+}
+
+export function parseQuery(query: Query): ParsedQuery {
```

Comment:
nit: the naming here is a bit misleading since the parsing is actually done prior to calling this function. Maybe `selectedTypesFromQuery` would be more appropriate?